File: paser/tools/git_tools.py
import subprocess
import re

def git_diff() -> str:
    try:
        result = subprocess.run(["git", "diff"], capture_output=True, text=True, check=True)
        return result.stdout if result.stdout else "No hay cambios."
    except subprocess.CalledProcessError as e:
        return f"Error: {e.stderr}"

def get_current_repo() -> str:
    try:
        result = subprocess.run(["git", "remote", "get-url", "origin"], capture_output=True, text=True, check=True)
        url = result.stdout.strip()
        match = re.search(r"[:/]([^/]+/[^/]+?)(?:\.git)?$", url)
        return match.group(1) if match else ""
    except Exception:
        return ""

File: src_js/tools/gitTools.js
import { exec } from 'child_process';
import { promisify } from 'util';

const execPromise = promisify(exec);

export const getCurrentRepo = async () => {
  try {
    const { stdout } = await execPromise('git remote get-url origin');
    const url = stdout.trim();
    const match = url.match(/[:\/]([^\/]+\/[^\/]+?)(?:\.git)?$/);
    return match ? match[1] : '';
  } catch (e) {
    return '';
  }
};